src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2166,195 +2166,53 @@ private ConsumeTokenResult ConsumeNextTokenUntilAfterAllCommentsAreSkippedMultiS
-        {
-            long prevTotalConsumed = _totalConsumed;
-            int i;
-            bool lastAsterisk = false;
-            while (true)
+                _tokenType = JsonTokenType.Comment;
+            }
+            else
             {
-                i = 0;
-                for (; i < localBuffer.Length; i++)
-                {
-                    byte nextByte = localBuffer[i];
-
-                    if (nextByte == JsonConstants.Slash && lastAsterisk)
-                    {
-                        goto Done;
-                    }
-
-                    if (nextByte == JsonConstants.Asterisk)
-                    {
-                        i++;
-                        lastAsterisk = true;
-                        if (i < localBuffer.Length)
-                        {
-                            if (localBuffer[i] == JsonConstants.Slash)
-                            {
-                                goto Done;
-                            }
-                        }
-                        else
-                        {
-                            if (!GetNextSpan())
-                            {
-                                _totalConsumed = prevTotalConsumed;
-                                return false;
-                            }
-                            _totalConsumed += localBuffer.Length + leftOver;
-                            _bytePositionInLine += localBuffer.Length + leftOver;
-                            leftOver = 0;
-                            localBuffer = _buffer;
-                            i = 0;
-                            if (localBuffer[i] == JsonConstants.Slash)
-                            {
-                                goto Done;
-                            }
-                            break;
-                        }
-                    }
-                    else if (nextByte == JsonConstants.LineFeed)
-                    {
-                        _bytePositionInLine = 0;
-                        _lineNumber++;
-                        lastAsterisk = false;
-                    }
-                    else
-                    {
-                        lastAsterisk = false;
-                    }
-                }
-                if (i == localBuffer.Length)
-                {
-                    if (!GetNextSpan())
-                    {
-                        _totalConsumed = prevTotalConsumed;
-                        return false;
-                    }
-                    _totalConsumed += localBuffer.Length + leftOver;
-                    _bytePositionInLine += localBuffer.Length + leftOver;
-                    leftOver = 0;
-                    localBuffer = _buffer;
-                }
+                _totalConsumed = prevTotalConsumed;
+                // consumed might not be correct now but it won't be used anymore after we return false
+                // if we don't wrong number of bytes might get reported (if consuming called GetNextSpan)
```

Comment:
I have a hard time understanding what this comment is saying.
`If we don't... wrong number`, if we don't do what?
And what do you mean by `consuming` in `if consuming called GetNextSpan`

Reply:
yep, re-worded with more details